.github/workflows/betelgeuse_test_case_validation_and_upload.yml

```diff
@@ -0,0 +1,74 @@
+---
+name: Betelgeuse Docstrings Validation and Automated Test Cases Upload
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    paths:
+      - "integration-tests/**"
+  schedule:
+    # Run weekly on Mondays at 2 AM UTC
+    - cron: '0 2 * * 1'
+  workflow_dispatch: # Allow manual triggering
+
+jobs:
+  betelgeuse-dry-run:
+    name: "betelgeuse dry-run"
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    container:
+      image: fedora:latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Base setup for Betelgeuse
+        run: |
+          dnf --setopt install_weak_deps=False install -y \
+              python3-pip
+          python3 -m pip install betelgeuse
+
+      - name: Run Betelgeuse
+        run: |
+          PYTHONPATH=integration-tests/ betelgeuse --config-module \
+            custom_betelgeuse_config test-case --dry-run \
+            integration-tests/ dryrun_project ./test_case.xml
```

Comment:
Since this is a dry run, would it make sense to expose the generated `./test_case.xml` as an artifact anyone could download and inspect?